File: Snooper/src/main/java/com/prateekj/snooper/networksnooper/views/HttpListView.java
package com.prateekj.snooper.networksnooper.views;

import com.prateekj.snooper.networksnooper.model.HttpCallRecord;

import java.util.List;

public interface HttpListView {
  void navigateToResponseBody(long httpCallId);

  void finishView();

  void showDeleteConfirmationDialog();

  void updateListViewAfterDelete();

  void initHttpCallRecordList(List<HttpCallRecord> httpCallRecords);

  void appendRecordList(List<HttpCallRecord> httpCallRecords);

  void renderNoCallsFoundView();
}

File: Snooper/src/main/java/com/prateekj/snooper/networksnooper/views/HttpListView.kt
package com.prateekj.snooper.networksnooper.views

import com.prateekj.snooper.networksnooper.model.HttpCallRecord

interface HttpListView {
  fun navigateToResponseBody(httpCallId: Long)
  fun finishView()
  fun showDeleteConfirmationDialog()
  fun updateListViewAfterDelete()
  fun initHttpCallRecordList(httpCallRecords: List<HttpCallRecord>)
  fun appendRecordList(httpCallRecords: List<HttpCallRecord>)
  fun renderNoCallsFoundView()
}
